### PR TITLE
JMPS module support

### PR DESCRIPTION
**Pull request by danlangford**
for kennycason/kumo#94

- added automatic module naming via the jar plugin in maven
- provided a way to easily config auto module name in each pom (module names cannot have '-' so using artifactId is sadly not allowed)
 - added `.api.` and `.core.` into some package names so they did not conflict. updated all references
- removed some logging dependencies that were introducing conflicting modules

this last piece here is something that may need some more work. the `kumo-core` pom was pulling in so many logging jars. when you have something like a `-core` that is used by other things like `-cli` and other peoples projects you really want to  just use the APIs and let the concrete projects determine their own logging method. however i did not spend a lot of time considering all of this. maybe another api is needed or adjustments to `-cli` is needed. 

the unit tests all pass.

### CODE REVIEW

File: kumo-core/pom.xml [hunk 1]
```diff
@@ -17,21 +21,9 @@
         </dependency>
 
         <!--logging -->
-        <dependency>
```
Comment (kennycason) [inline kumo-core/pom.xml:20]: thanks for cleaning this up! I have meant to update logging on this project forever now. :)

File: pom.xml [hunk 2]
```diff
@@ -240,6 +247,17 @@
                         </excludes>
                     </configuration>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-jar-plugin</artifactId>
+                    <configuration>
+                        <archive>
+                            <manifestEntries>
```
Comment (kennycason) [inline pom.xml:255]: Excellent.

**Review by kennycason — APPROVED** [review]
Looks great. Thanks a lot for the contribution! I will get this merged in, version bumped, and deployed to maven central soon. I have a few other updates to make and release as well. 

I'll shoot for this weekend, but please feel free to ping me if I don't for whatever reason.

**Comment by danlangford** [comment]
@kennycason just a little nudge to see how to version bump and release is going?

**Comment by Hafmaestro** [comment]
Hi @danlangford and @kennycason ,
I am working on a school project where we want to use Kumo but we can't until this PR has been merged, given conflicts with the package names. When do you think, @kennycason, that this can be merged?

Many thanks!

**Comment by kennycason** [comment]
@Hafmaestro sorry for falling off the grid. I have been recently overwhelmed with my current startup work + tech accelerator. 

I hope the delay didn't mess things up with you too badly. I'll just go ahead and merge and we'll go from there.

**Pull request merged**